ocaml/xapi/xapi_vbd_helpers.ml

```diff
@@ -74,30 +76,48 @@ let valid_operations ~expensive_sharing_checks ~__context record _ref' : table =
   let safe_to_parallelise = [`pause; `unpause] in
   (* Any current_operations preclude everything that isn't safe to parallelise *)
   ( if current_ops <> [] then
-      let concurrent_op = List.hd current_ops in
+      let concurrent_op_ref, concurrent_op_type = List.hd current_ops in
       set_errors Api_errors.other_operation_in_progress
-        ["VBD"; _ref; vbd_operations_to_string concurrent_op]
+        [
+          "VBD"
+        ; _ref
+        ; vbd_operations_to_string concurrent_op_type
+        ; concurrent_op_ref
+        ]
         (Listext.List.set_difference all_ops safe_to_parallelise)
   ) ;
   (* If not all operations are parallisable then preclude pause *)
-  let all_are_parallelisable =
-    List.fold_left ( && ) true
-      (List.map (fun op -> List.mem op safe_to_parallelise) current_ops)
+  let non_parallelisable_ops =
+    List.find_opt
+      (fun (_, op) -> not (List.mem op safe_to_parallelise))
+      current_ops
   in
   (* If not all are parallelisable, ban the otherwise
      parallelisable operations too *)
-  if not all_are_parallelisable then
-    set_errors Api_errors.other_operation_in_progress
-      ["VBD"; _ref; vbd_operations_to_string (List.hd current_ops)]
-      [`pause] ;
+  ( match non_parallelisable_ops with
+  | Some (concurrent_op_ref, concurrent_op_type) ->
+      set_errors Api_errors.other_operation_in_progress
+        [
+          "VBD"
+        ; _ref
+        ; vbd_operations_to_string concurrent_op_type
+        ; concurrent_op_ref
+        ]
+        [`pause]
+  | None ->
+      ()
+  ) ;
+
   (* If something other than `pause `unpause *and* `attach (for VM.reboot, see CA-24282) then disallow unpause *)
-  if
-    Listext.List.set_difference current_ops (`attach :: safe_to_parallelise)
-    <> []
-  then
-    set_errors Api_errors.other_operation_in_progress
-      ["VBD"; _ref; vbd_operations_to_string (List.hd current_ops)]
-      [`unpause] ;
+  let set_difference a b = List.filter (fun (_, x) -> not (List.mem x b)) a in
```

Comment:
`Listext.List.set_difference` is used further up ahead: you could replace all users, or keep using listext here/